crates/biome_js_analyze/src/lint/nursery/no_restricted_types.rs

```diff
@@ -0,0 +1,136 @@
+use crate::JsRuleAction;
+use ::serde::{Deserialize, Serialize};
+use biome_analyze::context::RuleContext;
+use biome_analyze::{
+    declare_lint_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_factory::make;
+use biome_js_syntax::{JsReferenceIdentifier, TextRange, TsReferenceType};
+use biome_rowan::BatchMutationExt;
+use biome_unicode_table::is_js_ident;
+use rustc_hash::FxHashMap;
+
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+
+declare_lint_rule! {
+    /// Disallow user defined types.
+    ///
+    /// This rule allows you to specify type names that you don’t want to use in your application.
+    ///
```

Comment:
We could add a link to the `noBannedTypes` rule and say that the rule handles types like `String` and `Function`.